andela/LIT-Kigali1
# #163519136 Add email notification on user signup

#### What does this PR do?
Adds email notification on user signup

#### Description of Task to be completed?
- Install @sengrid/mail package
- Create a mail controller
- Add test for mail controller
- Add mail notification after user signed up
- Implement email confirmation

#### How should this be manually tested?
- git clone this branch

- run `npm install`

- run `npm migrate:seed`

- run `npm run dev`

- with postman use this link to signup:
`POST: api/v1/users/`

- Body request example (login):
```request-json
{
   "user": {
       "username": "username",
       "email": "username@gmail.com"
       "password": "123456"
   }
}
```
- At last check your mail inbox for the confirmation email

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
`#163519136`

#### Screenshots (if appropriate)
- Confirmation Email Link sent to oemapp.dev@gmail.com
![image](https://user-images.githubusercontent.com/17350127/52698816-fd156a00-2f7c-11e9-9574-61f96aefd7b9.png)

- Email confirmed sent to oemapp.dev@gmail.com
![image](https://user-images.githubusercontent.com/17350127/52698865-19b1a200-2f7d-11e9-8151-f3084272b2eb.png)


#### Questions:
N/A
